### ISSUE TITLE
Endpoints unreachable from web browser

### ISSUE THREAD

**Issue by sfolador** [dingo/api#359]
Hi, 
Everything is working out of the box if we call the api endpoints using the "accept header". So, using curl we can call endpoints with v1 and v2. 
**In the configuration, the  `'strict'`  option is set to FALSE.**
How can we reach those endpoints from a web browser (i.e.: http://local.testingapi/v1/endpoint )?

Our routes:

```

Route::api(
   array("version" => "v1","prefix" => "v1"),
    function () {
  Route::get(
            'endpoint',
            function () {
                echo "api 1";
            }
        );    
}
);

Route::api(
    array("version" => "v2","prefix" => "v2"),
    function () {
        Route::get(
            'endpoint',
            function () {
                echo "api 2";
            }
        );

    }
);

```

If we try to reach `http://local.testingapi/v2/api` it doesn't work and we get a NotFoundHttpException.

We tried to add this to `filters.php`:

```
App::before(
    function ($request) {
        $path = explode('/', Request::path());
        if (in_array($path[0], Config::get('app.api_versions'))) {
            $request->headers->set('Accept', 'application/vnd.ourvendorname.' . $path[0] . '+json');
        }
    }
);
```

the problem is that if we call http://local.testingapi/v2/endpoint it doesn't work and we get an exception:

`BadRequestHttpException('The requested API version is invalid.');`

Long story short: 

- using the browser we can access only the default version of the api
- changing the default version of the api to v2 allows us to reach http://local.testingapi/v2/endpoint
- adding the before filter allows us to get a different exception

Do you have any advice?

**Comment by jasonlewis** [comment]
First up you shouldn't be using a prefix for your different versions. That's what the `Accept` header is for. Generally speaking the default version will always be the latest version of your API. That's so that when people view the API via the browser (should you want that) they'll see the latest version. Again, generally speaking, there's no need for a user to be able to see a previous (outdated) version of your API.

I suggest you drop the version prefix and use the `Accept` header as it was intended to be used. This way users who consume your API only need to adjust their header that's sent (which can usually be set globally) so it's easier for them to transition from one version to the next. The URI remains the same. That's how it should be.
